### ISSUE TITLE
Wrong number of layers while doing triplet training after softmax pre-training

### ISSUE THREAD

**Issue by Szyzapor** [philipperemy/deep-speaker#82]
Hi, I' ve got a question about triplet + softmax pre-training. When I am using only softmax or triplet training everything works perfectly, but when I am trying to do softmax pre-training and then triplet training I am seeing such an error:

_ValueError: You are trying to load a weight file containing 58 layers into a model with 57 layers._

How I should do pre-training to avoid this error? Or should I remove one of the layers from saved model? This is not clear for me.

Could you please explain how it should be done properly?

**Comment by philipperemy** [comment]
@Szyzapor that looks weird to me it should have worked nicely. The fact that it's expecting 58 instead of 57 is probably because we remove the softmax layer before pre-training with the triplets. Maybe:

```python
model.load_weights(filepath, by_name=True)
```

will work. If it's not already the case.

**Comment by philipperemy** [comment]
@Szyzapor when you have time please provide more details. I'll close it for now
